gcc/config/riscv/vectorization.md

```diff
@@ -5656,3 +5656,998 @@
 			   operands[2], operands[3]);
   DONE;
 })
+
+;; =========================================================================
+;; == Mixed arithmetic
+;; =========================================================================
+
+;; -------------------------------------------------------------------------
+;; ---- [INT,FP] General binary mixed arithmetic
+;; -------------------------------------------------------------------------
+;; Includes:
+;; - vwadd.vv
+;; - vwaddu.vv
+;; - vfwadd.vv
+;; - vwsub.vv
+;; - vwsubu.vv
+;; - vfwsub.vv
+;; - vwadd.vx
+;; - vwaddu.vx
+;; - vfwadd.vf
+;; - vwsub.vx
+;; - vwsubu.vx
+;; - vfwsub.vf
+;; - vwmul.vv
+;; - vwmulu.vv
+;; - vwmulsu.vv
+;; - vfwmul.vv
+;; - vwadd.wv
+;; - vwaddu.wv
+;; - vfwadd.wv
+;; - vwsub.wv
+;; - vwsubu.wv
+;; - vfwsub.wv
+;; - vwadd.wx
+;; - vwaddu.wx
+;; - vfwadd.wf
+;; - vwsub.wx
+;; - vwsubu.wx
+;; - vfwsub.wf
+;; - vwmul.vx
+;; - vwmulu.vx
+;; - vwmulsu.vx
+;; - vfwmul.vf
+;; -------------------------------------------------------------------------
+
+(define_expand "vec_convert_v<su><optab>v<mode><vw>"
+  [(set (match_operand:<VW> 0 "register_operand")
+    (plus_minus_mult:<VW>
+      (any_extend:<VW>
+        (match_operand:VWI 1 "register_operand"))
+      (any_extend:<VW>
+        (match_operand:VWI 2 "register_operand"))))]
```

Comment:
You can just write all `match_operand` only for those pattern you will always expand, that should save some build time for gcc.
```
  [(match_operand:<VW> 0 "register_operand")
   (match_operand:VWI 1 "register_operand")
   (match_operand:VWI 2 "register_operand")]
```

Reply:
> 

OK. I am gonna collect all cases to changed all of them as you suggested and make a single commit for you to review. Thanks~

Reply:
> You can just write all `match_operand` only for those pattern you will always expand, that should save some build time for gcc.
> 
> ```
>   [(match_operand:<VW> 0 "register_operand")
>    (match_operand:VWI 1 "register_operand")
>    (match_operand:VWI 2 "register_operand")]
> ```

OK. I am gonna collect all cases to changed all of them as you suggested and make a single commit for you to review. Thanks~

Reply:
> You can just write all `match_operand` only for those pattern you will always expand, that should save some build time for gcc.
> 
> ```
>   [(match_operand:<VW> 0 "register_operand")
>    (match_operand:VWI 1 "register_operand")
>    (match_operand:VWI 2 "register_operand")]
> ```

Hi, kito. I am trying to optimize the pattern as you suggested. But I have no idea how to do so.
For example:
(define_expand "vec_convert_v<su><optab>v<mode><vw>"
  [(set (match_operand:<VW> 0 "register_operand")
    (plus_minus_mult:<VW>
      (any_extend:<VW>
        (match_operand:VWI 1 "register_operand"))
      (any_extend:<VW>
        (match_operand:VWI 2 "register_operand"))))]

I used code iterator `plus_minus_mult` and `any_extend` to expand `<optab>` and `<su>. Do you want me remove the 
code iterators and split the pattern into several patterns?  Thank you~

Reply:
For those kind of pattern I think it's fine to keep that :)